### PR TITLE
Make sure container start doesn't make the DNS fields nil.

### LINKED ISSUE #18236: HostConfig.DNS can end up null when 'start' call contains HostConfig
#17779 attempted to ensure that `HostConfig.Dns` was never null, but you can still get into that situation by a partially-populated `HostConfig` in the `start` call.

I know specifying `HostConfig` at `start` is deprecated, but some 3rd-party tools are still doing it.

````
$ docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64
$ docker info
Containers: 17
Images: 102
Server Version: 1.9.1
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-21-generic
Operating System: Ubuntu 15.04
CPUs: 1
Total Memory: 363.1 MiB
Name: host1
ID: O3UY:UMAW:MIU3:3FEV:6XUL:UY4C:JCKU:IXYT:FWKB:VA5Q:W5ZO:NI36
WARNING: No swap limit support
$ uname -a
Linux host1 3.19.0-21-generic #21-Ubuntu SMP Sun Jun 14 18:31:11 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
````

Steps to Reproduce:

Starting with a Docker daemon listing on TCP (since my `curl` doesn't speak unix sockets):

````
$ docker create --name=cx alpine /bin/sh
553eff2d90a95e088439efecfcfea917781a6577446c1e36c1ee6127be56d334
$ curl -X POST -H Content-Type:application/json "http://127.0.0.1:2375/containers/cx/start" -d '{"NetworkMode": "host"}'
$ docker inspect -f {{.HostConfig.DNS}} cx
[]
$ docker inspect -f {{.HostConfig.Dns}} cx
Template parsing error: template: :1:13: executing "" at <.HostConfig.Dns>: Dns is not a field of struct type *runconfig.HostConfig
````

compare if you supply the same `HostConfig` at creation-time, you don't get the error:

````
$ docker rm -f cx
cx
$ docker create --net=host --name=cx alpine /bin/sh
6665e742c1715a5c7de11e2e3e1d1d5d57aa9d4b5c4308455c682f7fa237e23b
$ curl -X POST -H Content-Type:application/json "http://127.0.0.1:2375/containers/cx/start" -d ''
$ docker inspect -f {{.HostConfig.DNS}} cx
[]
$ docker inspect -f {{.HostConfig.Dns}} cx
[]
````

### PR DESCRIPTION
**Pull request by calavera**
Fixes #18236

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by calavera** [comment]
I added a log message in the daemon for this deprecated function. We already have a deprecation notice for 1.10 in https://github.com/docker/docker/blob/master/docs/misc/deprecated.md#hostconfig-at-api-container-start

File: daemon/start.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *runconfig.HostConf
 		// This is kept for backward compatibility - hostconfig should be passed when
 		// creating a container, not during start.
 		if hostConfig != nil {
+			logrus.Info("DEPRECATED: Setting host configuration options when the container starts is deprecated and will be removed in Docker 1.12")
```
Comment (cpuguy83) [inline daemon/start.go]: Warn?

Comment (calavera) [inline daemon/start.go]: changed

**Comment by cpuguy83** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**